metricbeat/module/zookeeper/mntr/data.go

```diff
@@ -86,8 +87,16 @@ func eventMapping(serverId string, response io.Reader, r mb.ReporterV2, logger *
 	}
 
 	// only exposed by the Leader
-	if _, ok := fullEvent["zk_followers"]; ok {
+	if hasFollowers, hasLearners := fullEvent["zk_followers"], fullEvent["zk_learners"]; hasFollowers != nil || hasLearners != nil {
```

Comment:
This oneliner plus the block inside it is not very idiomatic in go and `hasFollowers` is not a boolean variable, it's an int so its name should be something like `numFollowers` in this case or something along those lines.

Please, write each case separatedly and define the situations where you are omitting some result with a comment. If you can omit some value, please write a comment explaining why. Example:

```go
if value != 7 {
    // we can safely ignore values under same because reason 1 and values over 7 because reason 2. Decide if it's worth to write a debug log (not info) in such case in case things get hairy in the future
}
doLogicWithSeven(value)
```
You have partially done it already, I just can't quickly why learners and followers are related in the source code to be in the same block.

Reply:
It's just a style request, not saying it's incorrect btw.